inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -158,106 +158,183 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const std::string&
     _core = _multiPlugin->GetCore(); // shared_ptr that holds the Core
     _config[MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES] = strDevices;
 
-    std::vector<DeviceInformation> needLoadDevices;
-
-    // check if have cpu device
-    const auto CPUIter = std::find_if(metaDevices.begin(), metaDevices.end(),
-                                      [=](const DeviceInformation& d)->bool{return d.deviceName.find("CPU") != std::string::npos;});
-    if (CPUIter != metaDevices.end()) {
-        _cpuDevice = *CPUIter;
-        _config.insert(_cpuDevice.config.begin(), _cpuDevice.config.end());
-        needLoadDevices.push_back(_cpuDevice);
-        _cpuFuture = _cpuPromise.get_future();
-    }
-
-    // get accelerator device, like GPU
-    auto networkPrecision = GetNetworkPrecision(network);
-    _acceleratorDevice = _multiPlugin->SelectDevice(metaDevices, networkPrecision);
-    bool isAccelerator =
-        _acceleratorDevice.deviceName.find("CPU") == std::string::npos;
-    if (isAccelerator) {
-        _config.insert(_acceleratorDevice.config.begin(), _acceleratorDevice.config.end());
-        needLoadDevices.push_back(_acceleratorDevice);
-        _acceleratorFuture = _acceleratorPromise.get_future();
-    }
-
-    if (needLoadDevices.size() == 0) {
-        IE_THROW() << "No device set";
-    }
-
-    // will not wait for loading accelerator network,
-    // so the executor can't be destroyed before finished the task,
-    // so use executor as a member of MultiDeviceExecutableNetwork.
-    _executor = InferenceEngine::ExecutorManager::getInstance()->getIdleCPUStreamsExecutor(
-            IStreamsExecutor::Config{"AutoDeviceAsyncLoad",
-            static_cast<int>(std::thread::hardware_concurrency()) /* max possible #streams*/,
-            1 /*single thread per stream*/,
-            IStreamsExecutor::ThreadBindingType::NONE});
-
-    for (auto& p : needLoadDevices) {
-        // initialize these containers firstly to avoid insert operation in threads
-        _idleWorkerRequests[p.deviceName];
-        _workerRequests[p.deviceName];
-        _inferPipelineTasksDeviceSpecific[p.deviceName] = NULL;
-        const auto device = p.deviceName;
-        const auto deviceConfig = p.config;
+    // initialize the flags of load network context
```

Comment:
you may move that to the declaration of variables in the class, pls see below

Reply:
fix it